src/mpid/ch4/netmod/ofi/ofi_impl.h

```diff
@@ -707,20 +707,26 @@ MPL_STATIC_INLINE_PREFIX
         params.last_loc = 0;
         params.start_loc = 0;
         params.last_chunk = 0;
-        MPIR_Segment_init(NULL, 1, dt_datatype, &dt_seg);
-        MPIR_Segment_manipulate(&dt_seg, 0, &dt_size,
+        dt_seg = MPIR_Segment_alloc();
+        MPIR_Segment_init(NULL, 1, dt_datatype, dt_seg);
+        MPIR_Segment_manipulate(dt_seg, 0, &dt_size,
                                 MPIDI_OFI_contig_count_block,
                                 NULL, NULL, NULL, NULL, (void *) &params);
+        MPIR_Segment_free(dt_seg);
         count1 = params.count;
+
         params.count = 0;
         params.last_loc = 0;
         params.start_loc = 0;
         params.last_chunk = 0;
-        MPIR_Segment_init(NULL, dtc, dt_datatype, &dt_seg);
-        MPIR_Segment_manipulate(&dt_seg, 0, &dt_size,
+        dt_seg = MPIR_Segment_alloc();
+        MPIR_Segment_init(NULL, dtc, dt_datatype, dt_seg);
+        MPIR_Segment_manipulate(dt_seg, 0, &dt_size,
                                 MPIDI_OFI_contig_count_block,
                                 NULL, NULL, NULL, NULL, (void *) &params);
+        MPIR_Segment_free(dt_seg);
         count2 = params.count;
+
         if (count2 == 1) {      /* Contiguous */
```

Comment:
While line change

Reply:
Fixed locally.  Will push once the tests complete.